insecure/attacknetwork/corruptedConnector.go

```diff
@@ -52,6 +60,10 @@ func (c *CorruptedConnector) Connect(ctx context.Context, targetId flow.Identifi
 	return &CorruptedNodeConnection{stream: stream}, nil
 }
 
+func (c *CorruptedConnector) WithAttackerAddress(address string) {
```

Comment:
The short answer is: that the attacker address _may be_ determined post-startup rather than the initialization time. 

The long answer is: for testing, we use `localhost:0` as the attacker address so that the actual attacker port number will be picked by the kernel at the startup of the attacker. Hence, in testing, we may not have the correct port number of the attacker till it successfully starts up. This is to avoid port collision and flakey behavior during the test execution. Moreover, the `AttackNetwork` is mostly meant to be utilized as an integration test tool, hence a dynamic runtime port selection benefits the health of testing as well. Accordingly, this configuration is set to be independent of the constructor of the `CoruptedConnector`, so that the port number of the attacker is set when the attacker is set post-startup.